sorting/radix_sort2.cpp

```diff
@@ -0,0 +1,160 @@
+ *
+ *using count sort at each step, which is stable.
+ *stable => already sorted according to previous digits.
+*/
+
+// header files
+#include <iostream>  // for io operations
+#include <vector>    // for std::vector
+/**
+ * @namespace sorting
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace radix_sort
+ * @brief Functions for [Radix sort](https://en.wikipedia.org/wiki/Radix_sort)
+ * algorithm
+ */
+namespace radix_sort {
+/**
+ * @brief Class to create a vector and sort it.
+ */
+class radix {
+ public:
+    int n;
+    std::vector<int> ar;
+    /**
+     * @brief Constructor that initializes the vector on creation.
+     * @param a vector to be sorted.
+     */
+    explicit radix(const std::vector<int>& a) {
+        ar = a;
+        n = ar.size();
+    }
+    /**
+     * @brief Function that returns the max element.
+     * @returns none
+     */
+    int get_max() {
+        int mx = 0;
+        for (int i = mx = 0; i < n; i++) mx = std::max(mx, ar[i]);
+        return mx;
+    }
+    /**
+     * @brief Function that print the elements of vector.
+     * @returns none
+     */
+    void show() {
+        for (int i = 0; i < n; i++) std::cout << ar[i] << " ";
+        std::cout << "\n";
+    }
+    void radix_sort();
+    void step_ith(int dig);
+};
+/**
+ * @brief Function to sort vector according to current digit using stable
+ * sorting.
+ * @returns none
+ */
+void radix::step_ith(int cur_digit) {  // sorting according to current digit.
+    std::vector<int> position(10, 0);
+    for (int i = 0; i < n; i++) {
+        position[(ar[i] / cur_digit) %
+                 10]++;  // counting frequency of 0-9 at cur_digit.
+    }
+    int cur = 0;
+    for (int i = 0; i < 10; i++) {
+        int a = position[i];
+        position[i] = cur;  // assingning starting position of 0-9.
+        cur += a;
+    }
+    std::vector<int> temp(n);
+    for (int i = 0; i < n; i++) {
+        temp[position[(ar[i] / cur_digit) % 10]] =
+            ar[i];  // storing ar[i] in ar[i]'s cur_digit expected position of
+                    // this step.
+        position[(ar[i] / cur_digit) %
+                 10]++;  // incrementing ar[i]'s cur_digit position by 1, as
+                         // current place used by ar[i].
+    }
+    for (int i = 0; i < n; i++) {
+        ar[i] = temp[i];
+    }
+}
+/**
+ * @brief Function to sort vector digit by digit.
+ * @returns none
+ */
+void radix::radix_sort() {
+    int max_ele = get_max();
+    for (int i = 1; max_ele / i > 0;
+         i *= 10) {  // loop breaks when i > max_ele because no further digits
+                     // left to makes changes in array.
+        step_ith(i);
+    }
+}
+}  // namespace radix_sort
+}  // namespace sorting
+
+/**
+ * Function to test the above algorithm
+ * @returns none
+ */
+static void test1() {
+    std::vector<int> ar = {432, 234, 143, 332, 123};
+    sorting::radix_sort::radix obj(ar);
+    obj.radix_sort();
+    obj.show();
```

Comment:
a more convenient to use syntax would be
```
sorting::radix_sort(arr);
```
And this can be achieved by removig the wrapper class `radix` around `std::vector`.
object oriented design is a good feature to smplify complex code but if the code itself is simple then i think there is no need to make it complex. As in this case OOP is used to make the implementation a bit more complex, the wrapper around `std::vector` is just contaning the `std::vector` and it's size `n` which it self is taken care by `std::vector`, then why to make things complex.